src/main/java/com/epam/test_generator/services/TestResultService.java

```diff
@@ -0,0 +1,254 @@
+     *
+     * @param projectId identifier of {@link Project}
+     * @param suitResultDTOS List of {@link RawSuitResultDTO}.
+     * @param authentication current user principals
+     * @return {@link TestResult}
+     */
+    public TestResult saveResult(Long projectId,
+                                 List<RawSuitResultDTO> suitResultDTOS, Authentication authentication) {
+        int amountOfPassed = 0;
+        int amountOfSkipped = 0;
+        int amountOfFailed = 0;
+
+        final String executedBy = ((AuthenticatedUser) authentication.getPrincipal()).getEmail();
+
+        final TestResult testResult = getTestResult(projectId, executedBy);
+
+        SuitResult suitResult;
+
+        final List<SuitResult> suitResults = new ArrayList<>();
+
+        for (RawSuitResultDTO s : suitResultDTOS) {
+            suitResult = getSuitResult(projectId, s.getId());
+            final List<CaseResult> caseResults = new ArrayList<>();
+
+            for (RawCaseResultDTO c : s.getCaseResultDTOList()) {
+                CaseResult caseResult = getCaseResult(projectId, s.getId(), c.getId());
+                if (!Status.SKIPPED.equals(c.getStatus())) {
+
+                    final List<Status> stepsStatuses = getStatusesFromEachStepOfCase(c);
+                    final Status statusBy = evaluateStatusFrom(stepsStatuses);
+                    caseResult.setResult(statusBy);
+                } else {
+                    caseResult.setResult(Status.SKIPPED);
+                }
+                caseResult.setDuration(c.getDuration());
+                suitResult.setDuration(c.getDuration() + suitResult.getDuration());
+
+                final List<StepResult> collect = getStepResults(projectId, s, c);
+                caseResult.setSteps(collect);
+                caseResults.add(caseResult);
+            }
+
+            testResult.setDuration(testResult.getDuration() + suitResult.getDuration());
+
+            suitResult.setCaseResults(caseResults);
+
+            final List<Status> collect = getStatusesFromSuitResult(suitResult);
+            suitResult.setStatus(evaluateStatusFrom(collect));
+
+            switch (suitResult.getStatus()) {
+                case PASSED:
+                    amountOfPassed++;
+                    break;
+                case SKIPPED:
+                    amountOfSkipped++;
+                    break;
+                case FAILED:
+                    amountOfFailed++;
+                    break;
+            }
+
+            suitResults.add(suitResult);
+        }
+
+        testResult.setSuits(suitResults);
+
+        final List<Status> collect = getStatusesFromEachSuitOfTestResult(testResult);
+        testResult.setStatus(evaluateStatusFrom(collect));
+
+        testResult.setAmountOfPassed(amountOfPassed);
+        testResult.setAmountOfFailed(amountOfFailed);
+        testResult.setAmountOfSkipped(amountOfSkipped);
+
+        return testResultDAO.save(testResult);
```

Comment:
Why are you returning TestResult, but not use it anywhere. I looked through the code and found that all other update methods are void and don't return anything, but this method returns TestResult. Ithink that you need also make this method void or use the returning object.